pkg/manager/manager_tasks.go

```diff
@@ -191,10 +192,20 @@ func (s *singleton) Generate(sprites bool, previews bool, markers bool, transcod
 			return
 		}
 
+		galleries, err := qg.All()
+		if err != nil {
+			logger.Errorf("failed to get galleries for generate")
+			return
+		}
+
 		delta := utils.Btoi(sprites) + utils.Btoi(previews) + utils.Btoi(markers) + utils.Btoi(transcodes)
 		var wg sync.WaitGroup
 		s.Status.Progress = 0
-		total := len(scenes)
+		lenScenes := len(scenes)
+		total := lenScenes
+		if thumbnails {
+			total += len(galleries)
+		}
```

Comment:
The `qg.All()` call should be moved in here so that the query is only run if we're generating thumbnails.

Reply:
done